freeipa/src/main/java/com/sequenceiq/freeipa/service/filter/GcpNetworkFilterProvider.java

```diff
@@ -0,0 +1,42 @@
+package com.sequenceiq.freeipa.service.filter;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.common.mappable.CloudPlatform;
+import com.sequenceiq.freeipa.entity.Network;
+
+@Service
+public class GcpNetworkFilterProvider implements NetworkFilterProvider {
+
+    @Override
+    public Map<String, String> provide(Network network, String networkId, Collection<String> subnetIds) {
+        Map<String, String> filter = new HashMap<>();
+        if (network.getAttributes() != null && network.getAttributes().getMap() != null) {
+            Map<String, Object> attributes = network.getAttributes().getMap();
+            String sharedProjectId = (String) attributes.get("sharedProjectId");
+            filter.put("sharedProjectId", sharedProjectId);
```

Comment:
this constant at least should be refactored into a field to avoid duplication here